cts/cts-cli.in

```diff
@@ -1856,19 +1856,29 @@ for t in $tests; do
     grep -e '^\* \(Passed\|Failed\)' "$TMPFILE"
 done
 
+function print_or_remove_file() {
+
+  eval TMPFILE="\$TMPFILE_$1"
+  if [[ $(diff -wq $test_home/cli/regression.$1.exp "$TMPFILE") ]]; then
+    rm -f "$TMPFILE"
+  else
+    echo "    $TMPFILE"
+  fi
+}
+
 if [ $num_errors -ne 0 ]; then
-    echo "$num_errors tests failed; see output in:"
+    if [ $failed -ne 0 ]; then
+      echo "$num_errors tests failed; see output in:"
+    fi
```

Comment:
Here, we still want to print "$num_errors tests failed" even if $failed is 0 (we just don't want to print the "see output in:"). There are two ways a test can fail: it can return an unexpected exit status (in which case num_errors > 0), or it can produce unexpected output (in which case failed=1).

This might be more obvious if we structure the code like:

    if [ $num_errors -ne 0 ] && [ $failed - ne 0 ]; then
        ... tests failed both ways, show "tests failed" message with output ...
    elif [ $num_errors -ne 0 ]; then
        ... tests failed but output was fine, show "tests failed" message without output ...
    elif [ $failed -eq 1 ]; then
        ... tests passed but output was bad (current handling is fine) ...
    else
        ... tests succeeded both ways (current handling is fine) ...
    endif